File: front-end/src/app/models/task/task.model.ts
export class TaskModel {
    id: number;
    name: string;
    description: string;
    status: string | null;
  
    constructor(id: number, name: string, description: string, status: string | null) {
      this.id = id;
      this.name = name;
      this.description = description;
      this.status = status;
    }
  }
  

File: back-end/src/main/java/com/milestones/models/TaskModel.java
package com.milestones.models;

import jakarta.persistence.*;

@Entity
@Table(name = "task_model")
public class TaskModel {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String description;
    private String status;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public TaskModel(String name, String description, String status) {
        this.name = name;
        this.description = description;
        this.status = status;
    }

    public TaskModel() {
    }
}
